lib/Sema/SemaCast.cpp

```diff
@@ -2700,6 +2700,22 @@ void CastOperation::CheckCStyleCast(bool IsCheckedScope) {
       SrcExpr = ExprError();
       return;
     }
+
+    // Disallow cast from other Checked Pointer types to nt_arary_ptr because 
+    // the SrcType might not point to a NULL-terminated array.
+    if (DestType->isPointerType()) {
+      if (cast<PointerType>(DestType)->getKind() == CheckedPointerKind::NtArray) {
```

Comment:
Could you use the function `isCheckedPointerNtArrayType` for testing whether a type is or is not an nt_array_ptr type?

Reply:
I have updated the code as suggested.